manager/integration/tests/test_cluster_autoscaler.py

```diff
@@ -0,0 +1,389 @@
+import pytest
+
+import math
+import requests
+import time
+
+from common import apps_api  # NOQA
+from common import client  # NOQA
+from common import core_api  # NOQA
+from common import make_deployment_cpu_request  # NOQA
+from common import pod_make  # NOQA
+from common import volume_name  # NOQA
+
+from common import get_longhorn_api_client
+from common import get_self_host_id
+from common import update_setting
+
+from common import cleanup_all_backing_images
+from common import create_and_check_volume
+from common import create_and_wait_deployment
+from common import create_and_wait_pod
+from common import create_backing_image_with_matching_url
+from common import create_pv_for_volume
+from common import create_pvc_for_volume
+from common import create_pvc_spec
+from common import delete_and_wait_pod
+from common import generate_random_data
+from common import read_volume_data
+from common import wait_for_volume_detached
+from common import wait_for_volume_healthy
+from common import write_pod_volume_data
+
+from common import Ki
+
+from common import BACKING_IMAGE_NAME
+from common import BACKING_IMAGE_QCOW2_URL
+from common import K8S_CLUSTER_AUTOSCALER_EVICT_KEY
+from common import K8S_CLUSTER_AUTOSCALER_SCALE_DOWN_DISABLED_KEY
+from common import RETRY_AUTOSCALER_COUNTS
+from common import RETRY_AUTOSCALER_INTERVAL
+from common import SETTING_K8S_CLUSTER_AUTOSCALER_ENABLED
+from common import SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL
+
+from test_node import set_node_cordon
+
+CPU_REQUEST = 150
+
+
+@pytest.mark.skip(reason="require K8s cluster-autoscaler")
+@pytest.mark.cluster_autoscaler  # NOQA
+def test_cluster_autoscaler(client, core_api, apps_api, make_deployment_cpu_request, request):  # NOQA
+    """
+    Scenario: Test CA
+
+    Given Cluster with Kubernetes cluster-autoscaler.
+    And Longhorn installed.
+    And Set `kubernetes-cluster-autoscaler-enabled` to `true`.
+    And Create deployment with cpu request.
+
+    When Trigger CA to scale-up by increase deployment replicas.
+         (double the node number, not including host node)
+    Then Cluster should have double the node number.
+
+    When Trigger CA to scale-down by decrease deployment replicas.
+    Then Cluster should scale-down to original node number.
+    """
+    # Cleanup
+    def finalizer():
+        configure_node_scale_down(core_api, client.list_node(), disable="")
+        set_node_cordon(core_api, host_id, False)
+    request.addfinalizer(finalizer)
+
+    host_id = get_self_host_id()
+    host_node = client.by_id_node(host_id)
+    configure_node_scale_down(core_api, [host_node], disable="true")
+    set_node_cordon(core_api, host_id, True)
```

Comment:
Why not do the scaling down disabling and cordon for all existing/original nodes?